File: HotFixProject/app/src/main/java/com/toy/qiaodata/hotfixproject/MainActivity.java
package com.toy.qiaodata.hotfixproject;

import android.os.Bundle;
import android.os.PersistableBundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Toast;

import com.tencent.bugly.beta.Beta;

/**
 * Created by YANG on 2018/5/9.
 */

public class MainActivity extends AppCompatActivity {
    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        findViewById(R.id.btn_test).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String tag = "ABC";
//                String tag = null;
                tag.toLowerCase();
                Toast.makeText(getApplicationContext(), "BUG已经修复", Toast.LENGTH_SHORT).show();
            }
        });

        findViewById(R.id.btn_check_upgrade).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Beta.checkUpgrade();
            }
        });
    }
}

File: Kotlin/redbubble/src/main/java/com/toy/qiaodata/redbubble/MainActivity.kt
package com.toy.qiaodata.redbubble

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.MotionEvent
import com.toy.qiaodata.geometrylibrary.`object`.Circle

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val circle = Circle(10, 3, 4)

        val x = circle.getX(14.0)
        if (x != null) {
            Log.d("activity", "x1:${x[0]} + x2:${x[1]}")
        }
    }

    override fun onTouchEvent(event: MotionEvent): Boolean {
        Log.d("activity", "x:${event.x} + y:${event.y}")
        return super.onTouchEvent(event)
    }
}
